core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/items/MethodParameterAllowedAnnotationsBuildItem.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkiverse.langchain4j.deployment.items;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * All Ai service method parameters with annotation starting with allowedPrefix are allowed to be part of the prompt
+ * template.
+ */
+public final class MethodParameterAllowedAnnotationsBuildItem extends MultiBuildItem {
+
+    private final String allowedPrefix;
```

Comment:
Wouldn't it be better for the build item to use a `Predicate` for deciding?

Reply:
Looks more flexible indeed. Let's dig that path.

Reply:
👍🏽